src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/CSharpConvertForEachToLinqQueryProvider.cs

```diff
@@ -0,0 +1,301 @@
+                        var ifStatement = (IfStatementSyntax)current;
+                        if (ifStatement.Else == null)
+                        {
+                            convertingNodesBuilder.Add(new ExtendedSyntaxNode(ifStatement, currentLeadingTokens, Enumerable.Empty<SyntaxToken>()));
+                            currentLeadingTokens = new List<SyntaxToken>();
+                            // Proceed the loop with the nested statement.
+                            current = ifStatement.Statement;
+                            break;
+                        }
+                        else
+                        {
+                            statementsCannotBeConverted = new[] { current };
+                            break;
+                        }
+
+                    case SyntaxKind.LocalDeclarationStatement:
+                        // This is a situation with "var a = something;" s the innermost statements inside the loop.
+                        var localDeclaration = (LocalDeclarationStatementSyntax)current;
+                        if (localDeclaration.Declaration.Variables.All(variable => variable.Initializer != null))
+                        {
+                            // Prepare variable declarations to be converted into separate let clauses.
+                            ProcessLocalDeclarationStatement(localDeclaration);
+                            statementsCannotBeConverted = Enumerable.Empty<StatementSyntax>();
+                        }
+                        else
+                        {
+                            // As above, if there is an empty initializer, stop processing.
+                            statementsCannotBeConverted = new[] { current };
+                        }
+
+                        break;
+
+                    case SyntaxKind.EmptyStatement:
+                        // The innermost statement is an empty statement, stop processing
+                        // Example:
+                        // foreach(...)
+                        // {
+                        //    ;<- empty statement
+                        // }
+                        statementsCannotBeConverted = Enumerable.Empty<StatementSyntax>();
+                        break;
+
+                    default:
+                        // If no specific case found, stop processing.
+                        statementsCannotBeConverted = new[] { current };
+                        break;
+                }
+            }
+
+            // Trailing tokens are collected in the reverse order: from extrenal block down to internal ones. Reverse them.
+            trailingTokensBuilder.Reverse();
+
+            return new ForEachInfo<ForEachStatementSyntax, StatementSyntax>(forEachStatement, convertingNodesBuilder.ToImmutable(), identifiersBuilder.ToImmutable(), statementsCannotBeConverted.ToImmutableArray(), currentLeadingTokens.ToImmutableArray(), trailingTokensBuilder.ToImmutable());
```

Comment:
these super long lines are really annoying to review.  can you try to keep things to maybe under 120 cols so they look ok on github? Thanks! :)